core/src/main/java/com/datastax/oss/driver/internal/core/cql/Conversions.java

```diff
@@ -441,4 +450,71 @@ public static CoordinatorException toThrowable(
         return new ProtocolError(node, "Unknown error code: " + errorMessage.code);
     }
   }
+
+  /** Returns a common size for all kinds of Request implementations. */
+  static int minimumRequestSize(Request statement, DriverContext context) {
+    Preconditions.checkArgument(
+        context instanceof InternalDriverContext,
+        "DriverContext provided cannot be used to calculate statement's size");
+
+    /* Header and payload are common inside a Frame at the protocol level */
```

Comment:
Nit: could you avoid block-style implementation comments? They are annoying because they don't nest, e.g. if I try to enclose the whole method in a block-style comment, this one closes it and I get a compile error.